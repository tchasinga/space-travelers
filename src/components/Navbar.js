import logo from '../asset/planet.png';

export default function Navbar() {
  return (
    <header>
      <nav className="nav-section">
        <img src={logo} className="space-logo" alt="logo" />
      </nav>
    </header>
  );
}
