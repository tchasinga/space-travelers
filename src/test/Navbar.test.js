import { fireEvent, render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Navbar from '../components/Navbar';

describe('Navbar Should render as expected', () => {
  it('should be similar with the snapshot', () => {
    const output = render(
      <MemoryRouter>
        <Navbar />
      </MemoryRouter>,
    );
    expect(output).toMatchSnapshot();
  });

  it('Redirect to the Rocket or Home page when the Rocket Link is clicked', () => {
    const { getByText } = render(
      <MemoryRouter>
        <Navbar />
      </MemoryRouter>,
    );
    const rocket = getByText('Rocket');
    fireEvent.click(rocket);
    expect(window.location.pathname).toBe('/');
  });
});
