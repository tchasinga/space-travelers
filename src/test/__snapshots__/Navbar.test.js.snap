// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Should render as expected should be similar with the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header>
        <nav
          class="nav-section"
        >
          <div
            class="logHeader"
          >
            <img
              alt="logo"
              class="space-logo"
              src="planet.png"
            />
            <h1>
              Space Travlers' Hub
            </h1>
          </div>
          <div>
            <ul
              class="getNavDesigner"
            >
              <a
                aria-current="page"
                class="linksStyles   active active"
                href="/"
              >
                Rocket
              </a>
              <a
                class="linksStyles"
                href="/mission"
              >
                Mission
              </a>
              <span
                class="upbar"
              >
                 | 
              </span>
              <a
                class="linksStyles"
                href="/profile"
              >
                My-profile
              </a>
            </ul>
          </div>
        </nav>
      </header>
    </div>
  </body>,
  "container": <div>
    <header>
      <nav
        class="nav-section"
      >
        <div
          class="logHeader"
        >
          <img
            alt="logo"
            class="space-logo"
            src="planet.png"
          />
          <h1>
            Space Travlers' Hub
          </h1>
        </div>
        <div>
          <ul
            class="getNavDesigner"
          >
            <a
              aria-current="page"
              class="linksStyles   active active"
              href="/"
            >
              Rocket
            </a>
            <a
              class="linksStyles"
              href="/mission"
            >
              Mission
            </a>
            <span
              class="upbar"
            >
               | 
            </span>
            <a
              class="linksStyles"
              href="/profile"
            >
              My-profile
            </a>
          </ul>
        </div>
      </nav>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
